libs/librrgraph/src/io/rr_graph_reader.cpp

```diff
@@ -115,3 +133,85 @@ void load_rr_file(RRGraphBuilder* rr_graph_builder,
             read_rr_graph_name);
     }
 }
+
+static RREdgeId process_rr_edge_override(const std::string& line,
+                                         float& overridden_Tdel,
+                                         const RRGraphView& rr_graph) {
+    std::istringstream iss(line);
+    char ch;
+    RREdgeId edge_id;
+
+    if (std::isdigit(line[0])) {
+        // Line starts with an integer
+        int first;
+        iss >> first;
+        edge_id = (RREdgeId)first;
+    } else if (line[0] == '(') {
+        // Line starts with (first, second)
+        int first, second;
+        iss >> ch >> first >> ch >> second >> ch;
+
+        RRNodeId src_node_id = RRNodeId(first);
+        RRNodeId sink_node_id = RRNodeId(second);
+
+        edge_id = rr_graph.rr_nodes().edge_id(src_node_id, sink_node_id);
+
+        VTR_LOGV_ERROR(!edge_id.is_valid(),
+                       "Couldn't find an edge connecting node %d to node %d\n",
+                       src_node_id,
+                       sink_node_id);
+
+    } else {
+        VTR_LOG_ERROR("Invalid line format:  %s\n", line.c_str());
+    }
+
+    if (!(iss >> overridden_Tdel)) {
+        VTR_LOG_ERROR("Couldn't parse the overridden delay in this line:  %s\n", line.c_str());
+    }
+
+    return edge_id;
+}
+
+void load_rr_edge_delay_overrides(std::string_view filename,
+                                  RRGraphBuilder& rr_graph_builder,
+                                  const RRGraphView& rr_graph) {
+    std::ifstream file(filename.data());
+    VTR_LOGV_ERROR(!file, "Failed to open the RR edge override file: %s\n", filename.data());
+
+    std::unordered_map<t_rr_switch_inf, RRSwitchId, t_rr_switch_inf::Hasher> unique_switch_info;
+    for (const auto& [rr_sw_idx, sw] : rr_graph.rr_switch().pairs()) {
+        unique_switch_info.insert({sw, rr_sw_idx});
+    }
+
+    std::string line;
+    bool firstLine = true;
+
+    while (std::getline(file, line)) {
+        if (firstLine) {
+            if (line.empty() || line[0] != '#') {
+                VTR_LOG_ERROR("Error: First line must start with #\n");
```

Comment:
I'd remove this and just throw anything after a # away (make it a comment character).